kafka-daq-report/src/main.rs

```diff
@@ -228,36 +242,55 @@ async fn poll_kafka_msg(consumer: StreamConsumer, shared_data: SharedData) {
-                                        frame_number,
-                                        num_channels_present,
-                                        num_samples_in_first_channel,
-                                        is_num_samples_identical,
-                                    ));
+                                let id = data.digitizer_id();
+                                {
+                                    let mut logged_data = shared_data.lock().unwrap();
+                                    logged_data
+                                        .entry(id)
+                                        .and_modify(|d| {
+                                            d.num_msg_received += 1;
+                                            d.recent_msg_count += 1;
+
+                                            d.last_msg_timestamp = timestamp;
+
+                                            d.message_rate = d.recent_msg_count as f64
+                                                / recent_message_lifetime as f64;
+
+                                            let num_channels = match data.channels() {
+                                                Some(c) => c.len(),
+                                                None => 0,
+                                            };
+                                            if !d.has_num_channels_changed {
+                                                d.has_num_channels_changed =
+                                                    num_channels != d.num_channels_present;
+                                            }
+                                            d.num_channels_present = num_channels;
+                                            if !d.has_num_channels_changed {
+                                                d.has_num_samples_changed =
+                                                    num_samples_in_first_channel
+                                                        != d.num_samples_in_first_channel;
+                                            }
+                                            d.num_samples_in_first_channel =
+                                                num_samples_in_first_channel;
+                                            d.is_num_samples_identical = is_num_samples_identical;
+                                        })
+                                        .or_insert(DigitiserData::new(
+                                            timestamp,
+                                            frame_number,
+                                            num_channels_present,
+                                            num_samples_in_first_channel,
+                                            is_num_samples_identical,
+                                        ));
+                                };
+
+                                let shared_data = Arc::clone(&shared_data);
+                                thread::spawn(move || {
```

Comment:
Is there a reason for using `std::thread` instead of `tokio::task`?

Reply:
I was better able to understand the documentation for `std::thread` than I was for `tokio::task` so I opted on the using the former. I will switch to using `tokio::task` instead.